From b37a3f6bb815f8e7b7ae0023c853aadc1207e487 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Tue, 1 Apr 2025 15:45:03 +0100
Subject: [PATCH] dtoverlays: Disable kernel drivers for humidity sensor on
 Sense HATs

The Sense HAT library talks directly to the humidity sensor via
i2cdev.

The overlay has always defined the humidity sensor, but the relevant
kernel module wasn't being built until
https://github.com/raspberrypi/linux/pull/6093, applied to 6.9 and
above.

We now have a kernel driver claiming the I2C address, so userspace
can't talk directly to the sensor.

Disable the humidity sensor in the overlays

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/rpi-sense-overlay.dts    | 2 +-
 arch/arm/boot/dts/overlays/rpi-sense-v2-overlay.dts | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/arch/arm/boot/dts/overlays/rpi-sense-overlay.dts
+++ b/arch/arm/boot/dts/overlays/rpi-sense-overlay.dts
@@ -52,7 +52,7 @@
 			hts221-humid@5f {
 				compatible = "st,hts221-humid", "st,hts221";
 				reg = <0x5f>;
-				status = "okay";
+				status = "disabled";
 			};
 		};
 	};
--- a/arch/arm/boot/dts/overlays/rpi-sense-v2-overlay.dts
+++ b/arch/arm/boot/dts/overlays/rpi-sense-v2-overlay.dts
@@ -46,7 +46,7 @@
 			hts221-humid@5f {
 				compatible = "st,hts221-humid", "st,hts221";
 				reg = <0x5f>;
-				status = "okay";
+				status = "disabled";
 			};
 
 			lsm9ds1-accel@6a {
